plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployCommandHandler.java

```diff
@@ -37,15 +38,23 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
     try {
       IProject project = helper.getProject(event);
       if (project != null) {
-        final IProject project1 = project;
-        StandardDeployJob deploy = new StandardDeployJob(new ProjectToStagingExporter(), defaultStagingDir, project1);
+        String now = Long.toString(System.currentTimeMillis());
+        StandardDeployJob deploy =
+            new StandardDeployJob(new ExplodedWarPublisher(),
+                                  new StandardProjectStaging(),
+                                  new AppEngineProjectDeployer(),
+                                  getTempDir().append(now),
+                                  project);
         deploy.schedule();
```

Comment:
Why don't you make it another configurable item as to whether it should be cleaned up?  If it should be kept, then we can give the directory a better name, like `projectId-module-YYYYMMDDHHMMSS` so that's it's more clear.